pkg/ddl/table.go

```diff
@@ -1476,6 +1485,23 @@ func checkTableNotExists(d *ddlCtx, t *meta.Meta, schemaID int64, tableName stri
 	return checkTableNotExistsFromStore(t, schemaID, tableName)
 }
 
+func checkConstraintNamesNotExists(t *meta.Meta, schemaID int64, constrains []*model.ConstraintInfo) error {
+	tbInfos, err := t.ListTables(schemaID)
```

Comment:
Please do an early check and return nil `if len(contraints) == 0`. Otherwise this will loop over all tables in the schema for every CREATE TABLE even if there are no constraints.
Also notice the typo `constrains` -> `constraints` with a `t`.
```suggestion
	if len(constraints) == 0 {
		return nil
	}
	tbInfos, err := t.ListTables(schemaID)
```

Reply:
fix: https://github.com/pingcap/tidb/pull/48062/commits/5f9ffa4fba5a6e56b891f9cfa70fe8117167e221